scipy/optimize/_minimize.py

```diff
@@ -435,6 +435,9 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
         warn('Method %s cannot handle bounds.' % method,
              RuntimeWarning)
     # - callback
+    if callback is not None:
+        callback, _callback = lambda x: _callback(np.copy(x)), callback
```

Comment:
Nit: It took me some time to figure out what was going on in this code. It's a little syntactically tricky. It might be easier to read in each of these instances if you make the copying wrapper into a utility function with a good name.

```
import functools

def callback_with_copy(callback):
    @functools.wraps(callback)
    def new_callback(x):
        return callback(x.copy())
    return new_callback
...
    if callback is not None:
        callback = callback_with_copy(callback)
```